File: fork-runner/src/main/java/com/shazam/fork/injector/store/TestCaseStoreInjector.java
package com.shazam.fork.injector.store;

import com.shazam.fork.store.TestCaseStore;

public class TestCaseStoreInjector {
    private static final TestCaseStore INSTANCE = new TestCaseStore();

    public static TestCaseStore testCaseStore() {
        return INSTANCE;
    }
}

File: fork-runner/src/main/java/com/shazam/fork/injector/store/TestCaseStoreInjector.kt
package com.shazam.fork.injector.store

import com.shazam.fork.injector.GsonInjector.gson
import com.shazam.fork.injector.system.FileManagerInjector.fileManager
import com.shazam.fork.store.TestCaseStore

object TestCaseStoreInjector {
    private val INSTANCE by lazy { TestCaseStore(fileManager(), gson()) }

    @JvmStatic
    fun testCaseStore(): TestCaseStore {
        return INSTANCE
    }
}
